pink/rondis/string/db_operations.cc

```diff
@@ -614,3 +614,164 @@ int rondb_get_rondb_key(const NdbDictionary::Table *tab,
     }
     return 0;
 }
+
+#define RONDB_INSERT 2
+#define RONDB_UPDATE 1
+#define REG0 0
+#define REG1 1
+#define REG2 2
+#define REG3 3
+#define REG4 4
+#define REG5 5
+#define REG6 6
+#define REG7 7
+#define LABEL0 0
+void incr_key_row(std::string *response,
+                  Ndb *ndb,
+                  const NdbDictionary::Table *tab,
+                  NdbTransaction *trans,
+                  struct key_table *key_row) {
+
+    const NdbDictionary::Column *value_start_col = tab->getColumn("value_start");
+    const NdbDictionary::Column *tot_value_len_col = tab->getColumn("tot_value_len");
+
+    NdbOperation::OperationOptions opts;
+    std::memset(&opts, 0, sizeof(opts));
+    /**
+     * The mask specifies which columns is to be updated after the interpreter
+     * has finished. The values are set in the key_row.
+     * We have 7 columns, we will update tot_value_len in interpreter, same with
+     * value_start.
+     *
+     * The rest, redis_key, rondb_key, value_data_type, num_rows and expiry_date
+     * are updated through final update.
+     */
+
+    const Uint32 mask = 0x57;
+    const unsigned char *mask_ptr = (const unsigned char *)&mask;
+
+    // redis_key already set as this is the Primary key
+    key_row->null_bits = 1; // Set rondb_key to NULL, first NULL column
+    key_row->num_rows = 0;
+    key_row->value_data_type = 0;
+    key_row->expiry_date = 0;
+
+    /* Define the interpreted program */
+    Uint32 code_buffer[128];
+    NdbInterpretedCode code(tab, &code_buffer[0], sizeof(code_buffer));
+    code.load_const_u16(REG0, 4); //Memory offset 0
```

Comment:
You mean to write "Memory offset 4"?

Reply:
Yep, but again a duplicate comment that can be removed.